src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface ExecutionEnv {
```

Comment:
Can I double check on `ExecutionEnv`?

Assuming the following dev environment,
- Linux PC, _A_, running ONE-vscode, and has connection with one TV named _TV1_.
- another PC, _B_, that has connection with two more TVs named _TV2_ and _TV3_.

As far as I understand, 4 `ExecutionEnv` (for_TV1_,  _TV2_, _TV3_, and _B_) will be instantiated to use all TVs. Could you let me know if this is correct?

Reply:
I thought that could be 5.

in that case it will show DeviceView below

```
Local (PC A, EnvManager)
L HostPC (PC A, ExecutionEnv)
L TV1(ExecutrionEnv)
Remote (PC B, EnvManager)
L HostPC (PC B, ExecutionEnv)
L TV2(ExecutrionEnv)
L TV3(ExecutrionEnv)
```

In case `PC B`, it will handle as a remote connection

Reply:
Oh. I see. Can I have one more question?
what if PC A or B does _not_ have a toolchain installed? Is PC A or B still a ExecutionEnv?

Reply:
yes. just those are ExecutionEnv.

Reply:
> yes. just those are ExecutionEnv.

As the user timeline using ONE-vscode
  * (1) There was nothing in `target devices` view
  * (1)-1 Or, as default, `Local PC` is registered without any `ExecutionEnv` below.
  * (2) User can add `ExecutionEnv` by his/her manual registration step. 
  * (3) Let's assume user is going to register TV#2 which is wire-connected with PC2
  * (4) User selects remote PC as `EnvManager` and register the IP of PC2
  * (5) User selects `ExecutionEnv` type as TV and select TV2
  * (6) Then, new `EnvManager` PC2 will be appear in `Target Devices` view with `ExecutionEnv` TV2 as its child node.

I'm not sure how PC A and PC B still a ExecutionEnv..

Reply:
um... A little bit differ.

if `ExecutionEnv` inherited class added on, I thought those could be added automatically. 

for example, if user add a remote `ExecutionEnvManager`, then it automatically make `ExecutionEnv` search and add it.

Reply:
Aha, now I understand how `ExecutionEnv` are automatically registered. Thank you for comment.

AFAIU, as you said `it automatically make ExecutionEnv search and add it`, `EnvManager` will search for its child `ExecutionEnv` with the result of command from `wrapCommand()`.
If there is no toolchain installed in PC B, then IMHO the command should not return anything and PC B cannot registered as `ExecutionEnv`. 

(I think this discussion is a little bit out of scope for this PR. API itself doesn't matter for this conversation and it looks good to me.)

Reply:
So AFAIU, a device can have `ExecutionEnv` if
- the device has toolchain installed (having simulator)
- or the device can run commands, e.g., `infer`
- or the device has connection with other devices that have `ExecutionEnv`

How about adding such comment for `ExecutionEnv`?

Since I and @yunjayh does not understand `ExecutionEnv` clearly and I am afraid that other people would not understand clearly either.